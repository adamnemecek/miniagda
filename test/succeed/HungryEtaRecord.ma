-- 2012-02-07

-- a recursive unit type
-- 2013-03-30 must be a cofun since not SN
cofun Hungry : -(i : Size) -> Set
{ Hungry i = [j < i] -> Hungry j
}

fun D : [i : Size] -> Hungry i -> Set {}

-- Don't try this at home!
-- let unique [i : Size] (x, y : Hungry i) (d : D i x) : D i y = d
-- loops! because of infinite eta-expansion performed in equality testing
-- similar to recursive record problem
