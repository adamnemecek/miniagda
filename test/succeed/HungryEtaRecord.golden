--- opening "HungryEtaRecord.ma" ---
--- scope checking ---
--- type checking ---
type  Hungry : -(i : Size) -> Set
{ Hungry i = .[j < i] -> Hungry j
}
type  D : .[i : Size] -> Hungry i -> Set
{}
--- evaluating ---
--- closing "HungryEtaRecord.ma" ---
